lib/ansible/modules/identity/kinit.py

```diff
@@ -0,0 +1,256 @@
+                cache = ticket_caches[i].strip().split('\n')
+
+                self.cache_name = cache[0]
+
+                for ticket in cache[3:]:
+                    if 'krbtgt/' not in ticket:
+                        continue
+                    else:
+                        self.ticket = ticket
+                        break
+
+        return self.cache_name, self.ticket
+
+    def renew_ticket(self):
+        if self.max_ticket_expiry_datetime > self.ticket_expiry_datetime:
+            rc, err = self.generate_new_ticket()
+        else:
+            rc = -1
+            err = 'An existing ticket has yet to expire'
+        return rc, err
+
+    def generate_new_ticket(self):
+        cmd = shlex.split('{0} {1} -l {2}'.format(self.kinit, self.principal, self.lifetime))
+
+        rc, dummy, err = self.module.run_command(cmd, data=self.password, binary_data=False)
+        return rc, err
+
+    def destroy_tickets(self):
+        cmd = shlex.split('{0} -q'.format(self.kdestroy))
+
+        if self.cache_name is not None:
+            cmd.extend(shlex.split('-c {0}'.format(self.cache_name)))
+
+        rc, dummy, err = self.module.run_command(cmd)
+
+        self.cache_name = None
+        self.ticket = None
+
+        return rc, err
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            principal=dict(required=True, type='str'),
+            password=dict(required=False, type='str', no_log=True),
+            lifetime=dict(required=False, type='int', default=60),
+            state=dict(default='present', choices=['present', 'absent'])
+        ),
+        required_if=[
+            ['state', 'present', ['principal', 'password']]
+        ],
+        supports_check_mode=True,
+    )
+
+    kerberos_ticket = KerberosTicket(module)
+
+    cache_name, ticket = kerberos_ticket.get_existing_ticket()
+
+    if ticket is None:
+        if kerberos_ticket.state == 'present':
+            if module.check_mode:
+                module.exit_json(changed=True)
+
+            rc, err = kerberos_ticket.generate_new_ticket()
+
+            if rc != 0:
+                module.fail_json(msg='Failed to obtain a Kerberos ticket', rc=rc, err=err)
+            else:
+                module.exit_json(changed=True, msg='Obtained a Kerberos ticket', cache_name=cache_name)
+
+        elif kerberos_ticket.state == 'absent':
+            if module.check_mode:
+                module.exit_json(changed=False)
+
+            module.exit_json(changed=False, msg='No existing Kerberos tickets to destroy', cache_name=cache_name)
+
+    else:
+        if kerberos_ticket.state == 'present':
+            if module.check_mode:
+                if kerberos_ticket.max_ticket_expiry_datetime > kerberos_ticket.ticket_expiry_datetime:
+                    module.exit_json(changed=True)
+                else:
+                    module.exit_json(changed=False)
+
+            rc, err = kerberos_ticket.renew_ticket()
```

Comment:
I'd say that if there is still a ticket that has a validity of more than was requested, we don't need to try to renew and can just return a changed=False, msg='We already had a sufficient ticket'?

Reply:
That's true. I will update the module.